docs/spec/distribution/overview.md

```diff
@@ -2,71 +2,73 @@
 following rewards between validators and associated delegators:
- - multi-token fees to be socially distributed, 
- - proposer reward pool, 
- - inflated atom provisions, and
- - validator commission on all rewards earned by their delegators stake
+
+- multi-token fees to be socially distributed
+- proposer reward pool
+- inflated atom provisions
+- validator commission on all rewards earned by their delegators stake
 
 Fees are pooled within a global pool, as well as validator specific
 proposer-reward pools. The mechanisms used allow for validators and delegators
 to independently and lazily withdraw their rewards.  
 
-## Shortcomings 
+## Shortcomings
 
 As a part of the lazy computations, each delegator holds an accumulation term
 specific to each validator which is used to estimate what their approximate
-fair portion of tokens held in the global fee pool is owed to them. 
+fair portion of tokens held in the global fee pool is owed to them.
 
 ```
 entitlement = delegator-accumulation / all-delegators-accumulation
 ```
 
-Under the circumstance that there were constant and equal flow of incoming
+Under the circumstance that there were constant and equal flows of incoming
 reward tokens every block, this distribution mechanism would be equal to the
 active distribution (distribute individually to all delegators each block).
-However this is unrealistic so deviations from the active distribution will
+However, this is unrealistic so deviations from the active distribution will
 occur based on fluctuations of incoming reward tokens as well as timing of
-reward withdrawal by other delegators. 
+reward withdrawal by other delegators.
 
-If you happen to know that incoming rewards are about significantly move up,
+If you happen to know that incoming rewards are about to significantly go up,
 you are incentivized to not withdraw until after this event, increasing the
-worth of your existing _accum_.
+worth of your existing _accum_. See [#2764](https://github.com/cosmos/cosmos-sdk/issues/2764)
+for further details.
 
 ## Affect on Staking
 
 Charging commission on Atom provisions while also allowing for Atom-provisions
 to be auto-bonded (distributed directly to the validators bonded stake) is
-problematic within DPoS. Fundamentally these two mechanisms are mutually
-exclusive. If there are Atom commissions and auto-bonding Atoms, the portion
-of Atoms the reward distribution calculation would become very large as the Atom
-portion for each delegator would change each block making a withdrawal of rewards
-for a delegator require a calculation for every single block since the last
-withdrawal. In conclusion, we can only have Atom commission and unbonded atoms
+problematic within BPoS. Fundamentally, these two mechanisms are mutually
+exclusive. If there are both commissions and auto-bonding on the staking token,
+the amount of staking tokens each validator and delegator has would change each
+block, so we would have to iterate through all of them.
```

Comment:
I think this would be clearer/more-accurate for the final sentence in this paragraph: 

If both commission and auto-bonding mechanisms are simultaneously applied to the staking-token then the distribution of staking-tokens between any validator and its delegators will change with each block. This then necessitates a calculation for _each_ delegation records for _each_ block - which is considered computationally infeasible.

Reply:
Sure, I like that. Will update.